platform/security/sts/security-sts-attributequeryclaimshandler/src/test/java/org/codice/ddf/security/claims/attributequery/TestAttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,237 @@
+
+        assertThat(processedClaimCollection.get(0).getClaimType().toString(), is(equalTo("Role")));
+        assertThat(processedClaimCollection.get(0).getValues().get(0),
+                is(equalTo("Guest-hasMapping")));
+
+        assertThat(processedClaimCollection.get(1).getClaimType().toString(),
+                is(equalTo("NameIdentifier")));
+        assertThat(processedClaimCollection.get(1).getValues().get(0),
+                is(equalTo("Name-hasMapping")));
+
+        // Does not have an attribute mapping.
+        assertThat(processedClaimCollection.get(2).getClaimType().toString(), is(equalTo("Email")));
+        assertThat(processedClaimCollection.get(2).getValues().get(0), is(equalTo("email")));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesWithBadURI() throws Exception {
+        cannedResponse = Resources
+                .toString(Resources.getResource(getClass(), "/SAMLResponseBadAttribute.xml"),
+                        Charsets.UTF_8);
+        supportedClaims.add("Bad: URI");
+
+        assertThat(retrieveClaimValues().size(), is(equalTo(0)));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesNullResponse() {
+        responseState = "NullResponse";
+
+        assertThat(retrieveClaimValues().size(), is(equalTo(0)));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesInvalidResponseException() {
+        responseState = "Exception";
+
+        assertThat(retrieveClaimValues().size(), is(equalTo(0)));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesInvalidUsername() {
+        isValidUsername = false;
+
+        assertThat(retrieveClaimValues().size(), is(equalTo(0)));
+    }
+
+    @Test
+    public void testSupportedClaimsTypes() {
+        List<URI> supportedClaimTypes = attributeQueryClaimsHandler.getSupportedClaimTypes();
+
+        assertThat(supportedClaimTypes.size(), is(equalTo(3)));
+        assertThat(supportedClaimTypes.get(0).toString(), is(equalTo("Role")));
+        assertThat(supportedClaimTypes.get(1).toString(), is(equalTo("NameIdentifier")));
+        assertThat(supportedClaimTypes.get(2).toString(), is(equalTo("Email")));
+    }
+
+    @Test
+    public void testSupportedClaimsTypesWithBadURI() {
+        supportedClaims.add("Bad: URI");
+
+        assertThat(attributeQueryClaimsHandler.getSupportedClaimTypes().size(), is(equalTo(3)));
+    }
+
+    private ProcessedClaimCollection retrieveClaimValues() {
+        ClaimCollection claimCollection = new ClaimCollection();
+        Claim claim = new Claim();
+        try {
+            claim.setClaimType(
+                    new URI("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"));
+        } catch (URISyntaxException e) {
+            fail("Could not create URI.");
+        }
+        claimCollection.add(claim);
+        ClaimsParameters claimsParameters = mock(ClaimsParameters.class);
+        Principal principal = mock(Principal.class);
+
+        if (!isValidUsername) {
```

Comment:
It looks like you only use isValidUsername here. Might be better to have a parameter that takes in a flag rather than have a variable floating around that tests share to keep them independent of one another.